Bacon, porkbelly and supporting changes

dchieh222 opened the pull request:
<!-- HOW TO USE: Under each "#### Heading" below, enter information relevant to your pull request.
Leave the headings unless they don't apply to your PR, and remove the comment blocks (surrounded with <!–– and ––>) when you are done.

NOTE: Please grant permission for repository maintainers to edit your PR.  It is EXTREMELY common for PRs to be held up due to trivial changes being requested and the author being unavailable to make them.
-->

#### Summary

SUMMARY: Content " Adds porkbelly from swinelike creature harvest and food items processed from it: bacon & cooked fatty meat."

#### Purpose of change

At present, while pigs can be encountered and butchered, they never produce any meat which is usable for bacon. Therefore, recipes that require bacon can only use randomly generated Pre-Cataclysm bacon. This resolves the issue by creating a porkbelly drop from pigs which can then be processed into bacon, as well as creating porkbelly which can be used for other fatty meat purposes.

This also corrects the incorrectly high vitamin level of current bacon to match their real-life values(basically no vitamins whatsoever). 

#### Describe the solution

This allows bacon to be created through the following process:

```
1. Harvest of a pig which produces porkbelly.

2. Porkbelly can then be processed into raw curing meat 
with sugar & salt, along with a cooking 2 item. This represents 
a form of simple marinade. It takes 8 hours for this marinate to 
complete.

3. The raw curing meat can be activated to become raw cured 
meat at the end of 8 hours.

4. The player can process this into bacon at the end of this process 
directly via hand-smoking: this creates 3 charges of bacon and roughly 
approximates the amount of calories found in Pre-Cataclysm Bacon.

```
ALTERNATE:

```
4a. Alternatively, the player can smoke the bacon inside 
of a smoker. This produces "uncut bacon".

5b. This uncut bacon can be processed via a recipe to cut the 
bacon with a cut 1 instrument, and get 3 charges of standard 
bacon, roughly approximating the amount of calories found in 
Pre-Cataclysm Bacon.
```

This also allows porkbelly to be used alternatively:
1. Directly cooking it to create "cooked fatty meat" with a high caloric content(and negative health value). 
2. Using this "cooked fatty meat" as a sandwich ingredient to create a high calorie sandwich. 

#### Describe alternatives you've considered

There are issues with the current solution:

1. The smoker appears to only create a 1 to 1 item, so it cannot create three pieces of bacon for the one unit of "raw cured meat" put inside. This necessitates the awkward "uncut bacon" solution.

2. Furthermore, I've also attempted a "transform" action on the uncut bacon to transform it via cutting; however, there seems to be no way to require a CUT 1 object via transform, and the "transform" action appears to take any stack of "uncut bacon" and turn it into only 3 charges, wasting meat and defying reality.

3. I've considered creating a generic fatty meat item for use from future animals. This is possible, but upon review of information,  at least turkey bacon would require such significantly more processing that it cannot be casually replicated with the porkbelly bacon process. 

4. I've considered using "cooked meat" as the landing point of porkbelly, however, the NUTRIENT OVERRIDE appears to contravene the purpose of having a highly fatty piece of meat.

5. As an intermediate product, "uncut bacon" does not get many uses. Its only purpose appears to be cut, or to be eaten in its oversized form. Is this acceptable?

6. Without the use of nitrates or curing salts, this is qualitatively a different form of bacon than most Pre-Cataclysm bacon. Should this be considered as "homemade bacon" as a separation from standard bacon? 

7. Bacon can also be created with liquid smoke. Deluxe bacon may be double-smoked. None of this has been considered in this PR. Countearguments is that the main reason to double-smoke bacon is for the flavor, and since there's no flavor buff for using high enjoyment ingredients, this seems quite beside the point. Its possible that one could justify higher lifespan for double-smoked bacon. 

#### Testing

1) test pork belly generation from pigs - success

2) test pork belly generation from boars - success 

3) test pork belly generation from piglets - success

4) test pork belly generation from boarlets - success

5) test cooking porkbelly - success

6) test bacon creation via hand-smoking recipe - earlier success, should work still

7) test bacon creation via smoker process and cutting - success

8) test cooking cured/curing meat - success via recipe recognition

9) test sandwich machine with biscuit and cooked fatty meat - success

Suggestions for further testing:

Please check to make sure that porkbelly is appropriately listed as a component in all raw meat recipes, and what is the effect of natural rot upon the transformation process of curing meat.  


#### Additional context

Recipe for homemade bacon was drawn from various sites, including http://letsmakesomethingawesome.com/2011/03/home-cured-bacon-without-nitrates/ . Basically, bacon without curing salts.

Bacon vitamins(or lack of thereof) confirmed by online [nutritional information](https://www.kunzler.com/products/bacon/lower-sodium/attachment/lower-sodium-bacon-nfp/).

Porkbelly nutritional information pulled from [wikipedia](https://en.wikipedia.org/wiki/Pork_belly), and scaled up.

Porkbelly butcher information gathered from [Quora](https://www.quora.com/How-many-pounds-of-bacon-are-in-a-pig).

Project information and details on [Monday.](https://lichie.monday.com/boards/828842881/)

I was assisted in some description work by the github user wafflecloser in this PR. Please also credit him if this is merged.

Thank for your time!

dchieh222 committed:
Add files via upload

Edited to include length for crossbows. Data from wikipedia as well as via descriptions within.

dchieh222 committed:
Corrected linting as per autoformatter.

dchieh222 committed:
Update data/json/items/ranged/crossbows.json

Certainly.

Co-authored-by: ZhilkinSerg <ZhilkinSerg@users.noreply.github.com>

dchieh222 committed:
Update data/json/items/ranged/crossbows.json

Co-authored-by: ZhilkinSerg <ZhilkinSerg@users.noreply.github.com>

dchieh222 committed:
Update data/json/items/ranged/crossbows.json

Co-authored-by: ZhilkinSerg <ZhilkinSerg@users.noreply.github.com>

dchieh222 committed:
Update data/json/items/ranged/crossbows.json

Co-authored-by: ZhilkinSerg <ZhilkinSerg@users.noreply.github.com>

dchieh222 committed:
Update data/json/items/ranged/crossbows.json

Co-authored-by: ZhilkinSerg <ZhilkinSerg@users.noreply.github.com>

dchieh222 committed:
Update data/json/items/ranged/crossbows.json

Co-authored-by: ZhilkinSerg <ZhilkinSerg@users.noreply.github.com>

dchieh222 committed:
Merge pull request #1 from CleverRaven/master

updating with latest information

dchieh222 committed:
Test

dchieh222 committed:
Merge pull request #2 from dchieh222/dchieh222-patch-1-1

Test

dchieh222 committed:
Added harvest changes for swinelike animals

Added harvest changes for pigs, boars, piglets and boar cubs to reflect porkbelly gathering.

dchieh222 committed:
Merge pull request #3 from dchieh222/dchieh222-patch-3

Added harvest changes for swinelike animals

dchieh222 committed:
Changed harvest for swinelike creatures

Changes harvest to allow porkbelly from pigs, boars and their child versions.

dchieh222 committed:
Merge pull request #4 from dchieh222/dchieh222-patch-4

Changed harvest for swinelike creatures

dchieh222 committed:
Added porkbelly recipes

Added bacon and cooked fatty meat recipes.

dchieh222 committed:
Merge pull request #5 from dchieh222/dchieh222-patch-6-1

Added porkbelly recipes

dchieh222 committed:
Updated bacon

Bacon updated with appropriate flag for "SMOKED" and vitamins removed as per research.

dchieh222 committed:
Merge pull request #6 from dchieh222/dchieh222-patch-8

Updated bacon

dchieh222 committed:
Add files via upload

porkbelly should be available for recipes that use raw red meat, and cooked fatty meat should be available for recipes that just need cooked meat.

dchieh222 committed:
Merge pull request #7 from dchieh222/dchieh222-patch-9-2

Add files via upload

dchieh222 committed:
Ran autoformatter against harvest.json

This should correct linting issues.

LyleSY commented:
I see a couple of issues that Travis is shouting about. Descriptions need two spaces after periods, don't know why. Also it seems to be upset about how you are specifying spoil time. Please double check that you are copying the same format that everything else uses.
I'm excited to see this go in. Salted porkbelly would be way more valuable than gold post-Apocalypse

dchieh222 commented:
Thanks, I'll double-check - I used the autoformatter and attempted inasmuch as possible to mimic others so far.

dchieh222 committed:
Corrected styling deviations. 

Compare and confirm no more styling errors.

dchieh222 committed:
Merge pull request #8 from dchieh222/dchieh222-patch-11

Corrected styling deviations.

dchieh222 committed:
Updated pricing

Updated to reflect rare value of porkbelly and associated products.

dchieh222 committed:
Merge pull request #9 from dchieh222/dchieh222-patch-12

Updated pricing

dchieh222 committed:
The joys of linting

Fixing lint error.

dchieh222 committed:
Merge pull request #10 from dchieh222/dchieh222-patch-13

The joys of linting

dchieh222 committed:
Removing duplicate entry.

dchieh222 committed:
Merge pull request #11 from dchieh222/dchieh222-patch-14

Removing duplicate entry.

dchieh222 commented:
What else is failing on Travis? I'm checking my files, but I don't see the errors. Thanks in advance.

Fris0uman commented on data/json/items/comestibles/carnivore.json:
```diff
@@ -121,6 +121,82 @@
+    "type": "COMESTIBLE",
+    "copy-from": "meat_cooked",
+    "name": { "str": "large slab of uncut bacon", "str_pl": "large slabs of uncut bacon" },
+    "description": "A huge slab of uncut smoked bacon. You can eat as is, but it would be more efficient to cut it into smaller portions with a knife.",
```
```suggestion
    "description": "A huge slab of uncut smoked bacon.  You can eat as is, but it would be more efficient to cut it into smaller portions with a knife.",
```

dchieh222 committed:
Update data/json/items/comestibles/carnivore.json

Thank you!

Co-authored-by: Fris0uman <41293484+Fris0uman@users.noreply.github.com>

dchieh222 committed:
Edits to caloric value

adding new calorie values, keeping syntax and style

dchieh222 committed:
Merge pull request #12 from dchieh222/dchieh222-patch-16-1

Edits to caloric value

dchieh222 committed:
linting

linting again

dchieh222 committed:
Merge pull request #13 from dchieh222/dchieh222-patch-17

linting

LyleSY commented:
This Travis test is still failing. I don't know enough about the test and its purpose to share insight on it though
![image](https://user-images.githubusercontent.com/26608431/98301766-50b5bb80-1f89-11eb-8f90-adff0ae1879f.png)

dchieh222 committed:
Add files via upload

Calories for raw curing fatty meat were far below what they should have been in the default.

dchieh222 committed:
Merge pull request #14 from dchieh222/dchieh222-patch-18-1

Add files via upload

dchieh222 commented:
Travis wants to add sugar calories to porkbelly and doesn't understand why I dropped the calories after processing without using Nutrient Override, I guess. Fixing.

dchieh222 commented:
I can't see any other failures - what am I missing?

Fris0uman commented:
Failure looks unrelated to your changes, so you should be good

dchieh222 commented:
I yams danielchieh on discord, with a sakura avatar that needs to be fixed

dchieh222 commented:
dchieh5656
